pkg/controller/restore_controller.go

```diff
@@ -253,15 +242,15 @@ func (c *restoreController) processRestore(restore *api.Restore) error {
 	}
 
 	// patch to update status and persist to API
-	updatedRestore, err := patchRestore(original, restore, c.restoreClient)
+	err := kubeutil.PatchResource(original, restore, c.kbClient)
 	if err != nil {
 		// return the error so the restore can be re-processed; it's currently
 		// still in phase = New.
 		return errors.Wrapf(err, "error updating Restore phase to %s", restore.Status.Phase)
 	}
 	// store ref to just-updated item for creating patch
-	original = updatedRestore
-	restore = updatedRestore.DeepCopy()
+	original = restore
+	restore = restore.DeepCopy()
```

Comment:
Does it need to DeepCopy restore itself again?

Reply:
This is used for the later patch function. The patch function needs to know the original and the altered versions to work.

``` go
	if err = kubeutil.PatchResource(original, restore, c.kbClient); err != nil {
		c.logger.WithError(errors.WithStack(err)).Info("Error updating restore's final status")
	}
```

Reply:
Changed to 
```
original = restore.DeepCopy()
```